plugin/pkg/shoot/seedmanager/admission.go

```diff
@@ -174,11 +197,14 @@ func determineSeed(shoot *garden.Shoot, seedLister gardenlisters.SeedLister, sho
 		candidates []*garden.Seed
 	)
 
-	// Determine all candidate seed cluster matching the shoot's cloud and region.
-	for _, seed := range seedList {
-		if seed.DeletionTimestamp == nil && seed.Spec.Cloud.Profile == shoot.Spec.Cloud.Profile && seed.Spec.Cloud.Region == shoot.Spec.Cloud.Region && seed.Spec.Visible != nil && *seed.Spec.Visible && verifySeedAvailability(seed) {
-			candidates = append(candidates, seed)
-		}
+	switch strategy {
+	case seedmanagerapi.SameRegion:
+		candidates = determineCandidatesWithSameRegionStrategy(seedList, shoot, candidates)
+		break
+	case seedmanagerapi.MinimalDistance:
+		candidates = determineCandidatesWithMinimalDistanceStrategy(seedList, shoot, candidates)
+	default:
+		candidates = determineCandidatesWithSameRegionStrategy(seedList, shoot, candidates)
```

Comment:
Let's rather return an error here because of an unknown strategy instead of doing something that is unexpected for the user.